crates/ra_hir_def/src/body/lower.rs

```diff
@@ -226,6 +265,7 @@ impl ExprCollector<'_> {
                         None => self.collect_expr_opt(condition.expr()),
                         // if let -- desugar to match
                         Some(pat) => {
+                            // FIXME(pfaria) track the break and arms parents here?
                             mark::hit!(infer_resolve_while_let);
                             let pat = self.collect_pat(pat);
                             let match_expr = self.collect_expr_opt(condition.expr());
```

Comment:
When collecting for expr children within this section (up to the return below) should I include the inner ones created with `alloc_expr_desugared`?

Reply:
@matklad or @flodiebold any idea what I should do in this section?

Reply:
That `debug_assert` was a good idea! I've got most everything else covered now except for one test that panics because of adding that. Can't finish tonight, but will try pushing something out tomorrow morning.

Reply:
Is the current `debug_assert` sufficient? I'm checking that all children found during walk exist in the map and that the counts exactly match how many children map to parents. Should I add something that logs which child ids are extra/missing?